File: Data Structures & Algorithms/sort-an-array/submission-0.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        n = len(nums)

        for i in range(n-1):
            swapped = False
            for j in range(n-1-i):
                if nums[j]>nums[j+1]:
                    temp = nums[j];
                    nums[j] = nums[j+1]
                    nums[j+1] = temp
                    swapped = True

            if not swapped:
                break
        return nums

File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {
        quickSort(nums, 0, nums.length - 1);
    
        
    }

    public static void quickSort(int[] nums, int low, int high){
        if (low<high){
            int pivotIndex = partition(nums, low, high);
            quickSort(nums, low, pivotIndex-1);
            quickSort(nums, pivotIndex+1,high);
        }
    }

    public static int partition(int[] nums, int low, int high){

        int pivot = nums[low];
        int i = low+1;
        int j = high;

        while(i <= j){
            while(i <=j && nums[i]<= pivot){
                i++;
            }

            while(i<=j && nums[j]> pivot){
                j--;
            }
            if(i<j){
                swap(nums, i,j);
            }
        }
        swap(nums, low, j);
        return j;

    }

        private static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
}